### PR TITLE
feat: improve github action / remote-building behavior

### PR DESCRIPTION
**Pull request by steveej**

### CODE REVIEW

File: modules/nixosConfigurations.linux-01/github-runner-multi-arch.nix [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
-      extraLabels = ["multi-arch"];
-      tokenFile = (import ./magicPaths.nix).githubRunnerHraTokenHostPath;
+      ephemeral = true;
+      extraLabels = [ githubRunnersCfg.namePrefix ];
+      tokenFile = magicPaths.githubRunnerHraTokenHostPath;
       url = "https://github.com/holochain/holochain";
       extraPackages = config.environment.systemPackages;
     });
+
+  nix.settings.trusted-users = (mkList "github-runner-");
```
Comment (DavHau) [inline modules/nixosConfigurations.linux-01/github-runner-multi-arch.nix:25]: Why Is this needed? It's probably netter if the user is not trusted

File: modules/nixosConfigurations.linux-01/github-runner-multi-arch.nix [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
+  githubRunnersCfg = {
+    count = 10;
+    namePrefix = "multi-arch";
+  };
+
+  mkList = prefix: (builtins.genList
+    (x: "${prefix}${githubRunnersCfg.namePrefix}-${builtins.toString x}")
+    githubRunnersCfg.count);
+
+in
```
Comment (DavHau) [inline modules/nixosConfigurations.linux-01/github-runner-multi-arch.nix:12]: Might be a nit pick, but I think it's often better to keep abstractions in nixos configs at a minimum and only use it where it's really necessary.
Extra abstractions always come at a cost. They increases the mental capacity required to understand the code and make it harder to grasp what's happening, especially for less experienced nix users.

File: shared.nix [hunk 3]
```diff
@@ -8,7 +8,9 @@
 
   nix.settings.builders-use-substitutes = true;
 
-  nix.settings.max-jobs = lib.mkDefault 4;
+  nix.settings.max-jobs = lib.mkDefault "auto";
+  nix.settings.keep-outputs = true; # Nice for developers
+  nix.settings.keep-derivations = true; # Idem
```
Comment (DavHau) [inline shared.nix:13]: Why do we want this on a server ?

File: modules/_template.nix [hunk 4]
```diff
@@ -1,8 +1,4 @@
 { self, lib, ... }: {
-  perSystem = { config, self', inputs', pkgs, ... }: {
-
-  };
-  flake = {
-
-  };
+  perSystem = { config, self', inputs', pkgs, ... }: { };
+  flake = { };
```
Comment (DavHau) [inline modules/_template.nix:3]: formatting the template like this makes it more confusing I believe.

Comment (steveej) [reply modules/_template.nix:3]: just a result of formatting all nix files in one generic command

File: shared.nix (hunk 3 above)
Comment (steveej) [reply shared.nix:13]: it's an experiment to see if it resolves errors about invalid drv store paths in the build logs

File: modules/nixosConfigurations.linux-01/github-runner-multi-arch.nix (hunk 1 above)
Comment (steveej) [reply modules/nixosConfigurations.linux-01/github-runner-multi-arch.nix:25]: i wanted the cachix action to succeed to see whether it resolves cachix errors

File: modules/_template.nix (hunk 4 above)
Comment (steveeJ) [reply modules/_template.nix:3]: ideally we change the cachix action to tolerate running with an untrusted user. the issue i've seen is that the file that captures the current store paths isn't created, which leads to it not being found when attempting to diff and push.

File: modules/nixosConfigurations.linux-01/github-runner-multi-arch.nix (hunk 1 above)
Comment (steveej) [reply modules/nixosConfigurations.linux-01/github-runner-multi-arch.nix:25]: ideally we change the cachix action to tolerate running with an untrusted user. the issue i've seen is that the file that captures the current store paths isn't created, which leads to it not being found when attempting to diff and push.

Comment (steveej) [reply modules/nixosConfigurations.linux-01/github-runner-multi-arch.nix:25]: by now i strongly suspect that the invalid paths issues i've seen is due to garbage-collection cleaning up the paths before they can get pushed.

(besides that it probably also breaks builds)
